### PR TITLE
"More" fluent API and more true with F#

### PR DESCRIPTION
**Pull request by mastoj**
I updated the api so you can "pipe" basically everything. There is more work to do if you also want validation, but that wasn't in the old version so I didn't add now either.

Haven't used rake before, so when I set everything up there were some changes to sln etc. hope that doesn't mess things up.

### CODE REVIEW

File: src/Topshelf.FSharp.Example/Topshelf.FSharp.Example.fsproj [hunk 1]
```diff
@@ -64,7 +65,19 @@
   <PropertyGroup>
     <MinimumVisualStudioVersion Condition="'$(MinimumVisualStudioVersion)' == ''">11</MinimumVisualStudioVersion>
   </PropertyGroup>
-  <Import Project="$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.0\Framework\v4.0\Microsoft.FSharp.Targets" Condition=" Exists('$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.0\Framework\v4.0\Microsoft.FSharp.Targets')" />
+  <Choose>
+    <When Condition="'$(VisualStudioVersion)' == '11.0'">
+      <PropertyGroup>
+        <FSharpTargetsPath>$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.0\Framework\v4.0\Microsoft.FSharp.Targets</FSharpTargetsPath>
```
Comment (haf) [inline src/Topshelf.FSharp.Example/Topshelf.FSharp.Example.fsproj:71]: Should be v3.1

File: src/Topshelf.FSharp/Topshelf.FSharp.fsproj [hunk 2]
```diff
@@ -35,14 +36,26 @@
   <PropertyGroup>
     <MinimumVisualStudioVersion Condition="'$(MinimumVisualStudioVersion)' == ''">11</MinimumVisualStudioVersion>
   </PropertyGroup>
-  <Import Project="$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.0\Framework\v4.0\Microsoft.FSharp.Targets" Condition=" Exists('$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.0\Framework\v4.0\Microsoft.FSharp.Targets')" />
+  <Choose>
+    <When Condition="'$(VisualStudioVersion)' == '11.0'">
+      <PropertyGroup>
+        <FSharpTargetsPath>$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.0\Framework\v4.0\Microsoft.FSharp.Targets</FSharpTargetsPath>
```
Comment (haf) [inline src/Topshelf.FSharp/Topshelf.FSharp.fsproj:42]: Same here?

**Comment by haf** [comment]
Nice :).

Will merge – what do you think about the SDK versions?

File: src/Topshelf.FSharp.Example/Topshelf.FSharp.Example.fsproj (hunk 1 above)
Comment (mastoj) [inline src/Topshelf.FSharp.Example/Topshelf.FSharp.Example.fsproj:71]: Yeah I can fix if I get more time later. On my to a wedding so limited time this weekend. 

Opened it in VS 15, that's probably why I got that version.

-----Original Message-----

**Comment by mastoj** [comment]
I didn't mean to do anything with the versions, it was VS that forced my and that it was i got when opening it in VS15. VS13 also did some updates so I just went with it to get things done instead :)

**Comment by haf** [comment]
Alright.

**Comment by haf** [comment]
Released v2.0.1 https://www.nuget.org/packages/Topshelf.FSharp/

Thanks for your work!

**Comment by mastoj** [comment]
@haf no problem :) But was the previous version 2.0.0? If that was the case I think you should have updated the api to 3.0.0 since it was breaking changes all over the place, I mean if someone is using the package :)

**Comment by mastoj** [comment]
@haf now I see you created 2.0.0 and 2.0.1 from this PR :). :+1:

**Pull request merged**